### ISSUE TITLE
java.lang.IllegalArgumentException: surfaceTexture must not be null

### ISSUE THREAD

**Issue by shoebhoney** [izacus/AndroidOpenGLVideoDemo#3]
Hello Developer,

Great Work!

But i am getting this error.

```
java.lang.IllegalArgumentException: surfaceTexture must not be null
                                                                                               at android.view.Surface.<init>(Surface.java:140)
                                                                                               at compassapp.tv.com.videoeffectsdemobyshoeb.MainActivity.startPlaying(MainActivity.java:55)
                                                                                               at compassapp.tv.com.videoeffectsdemobyshoeb.MainActivity.onSurfaceTextureAvailable(MainActivity.java:70)
                                                                                               at android.view.TextureView.getHardwareLayer(TextureView.java:368)
                                                                                               at android.view.View.updateDisplayListIfDirty(View.java:15157)
                                                                                               at android.view.View.draw(View.java:15954)
                                                                                               at android.view.ViewGroup.drawChild(ViewGroup.java:3609)
                                                                                               at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3399)
                                                                                               at android.view.View.updateDisplayListIfDirty(View.java:15175)
                                                                                               at android.view.View.draw(View.java:15954)
                                                                                               at android.view.ViewGroup.drawChild(ViewGroup.java:3609)
                                                                                               at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3399)
                                                                                               at android.view.View.updateDisplayListIfDirty(View.java:15175)
                                                                                               at android.view.View.draw(View.java:15954)
                                                                                               at android.view.ViewGroup.drawChild(ViewGroup.java:3609)
                                                                                               at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3399)
                                                                                               at android.view.View.updateDisplayListIfDirty(View.java:15175)
                                                                                               at android.view.View.draw(View.java:15954)
                                                                                               at android.view.ViewGroup.drawChild(ViewGroup.java:3609)
                                                                                               at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3399)
                                                                                               at android.view.View.updateDisplayListIfDirty(View.java:15175)
                                                                                               at android.view.View.draw(View.java:15954)
                                                                                               at android.view.ViewGroup.drawChild(ViewGroup.java:3609)
                                                                                               at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3399)
                                                                                               at android.view.View.updateDisplayListIfDirty(View.java:15175)
                                                                                               at android.view.View.draw(View.java:15954)
                                                                                               at android.view.ViewGroup.drawChild(ViewGroup.java:3609)
                                                                                               at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3399)
                                                                                               at android.view.View.draw(View.java:16187)
                                                                                               at com.android.internal.policy.PhoneWindow$DecorView.draw(PhoneWindow.java:2690)
                                                                                               at android.view.View.updateDisplayListIfDirty(View.java:15180)
                                                                                               at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:281)
                                                                                               at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:287)
                                                                                               at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:322)
                                                                                               at android.view.ViewRootImpl.draw(ViewRootImpl.java:2615)
                                                                                               at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2434)
                                                                                               at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2067)
                                                                                               at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1107)
                                                                                               at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6013)
                                                                                               at android.view.Choreographer$CallbackRecord.run(Choreographer.java:858)
                                                                                               at android.view.Choreographer.doCallbacks(Choreographer.java:670)
                                                                                               at android.view.Choreographer.doFrame(Choreographer.java:606)
                                                                                               at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
                                                                                               at android.os.Handler.handleCallback(Handler.java:739)
                                                                                               at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                                               at android.os.Looper.loop(Looper.java:148)
                                                                                               at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                                                               at java.lang.reflect.Method.invoke(Native Method)
                                                                                               at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                                                               at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

**Comment by martoriro** [comment]
For anyone wondering. This error ocurrs because the main thread tries to access the video texture (renderer.getVideoTexture()) before it's set (videoTexture is set on setupTexture, inside initGLComponents method, which is executed by the thread that is initialized on VideoTextureRenderer instantiaton).

A solution for this is splitting startPlaying method on one that's called when the surfaceTexture is available (as the tutorial says) and that instantiates the VideoTextureRenderer object, and a callback method that sets the player attributes (all that's inside the try and catch) and it's called on the thread at the end of initGLComponents method.

**Comment by izacus** [comment]
Fixed in ef90060535f8778652e1e42fd0d5ea0e14707b86 :)
